src/test/java/com/gluonhq/substrate/util/FileOpsTests.java

```diff
@@ -216,4 +221,50 @@ void parseNonXMLFile() throws IOException {
         assertThrows(IOException.class, () -> FileOps.getNodeValue(resourcePath.toString(), "aa", "bb"));
         Files.deleteIfExists(resourcePath);
     }
+    
+    @Test
+    void removeEmptyLinesFromFile() throws IOException {
+        Path tempFile = Files.createTempFile("test", String.valueOf(System.currentTimeMillis()));
+        Files.writeString(tempFile, "abc\n    \nxyz");
+        List<String> strings = Files.readAllLines(tempFile);
+        assertEquals(3, strings.size());
+        FileOps.removeEmptyLines(tempFile);
+        strings = Files.readAllLines(tempFile);
+        assertEquals(2, strings.size());
+        Files.deleteIfExists(tempFile);
+    }
+
+    @Test
+    void testNodeAddition() throws IOException {
+        Path xmlPath = FileOps.copyResource("/test-ops.xml", getTempDir().resolve("test-ops.xml"));
+        System.out.println(Files.readString(xmlPath));
+        assertTrue(Files.exists(xmlPath));
+        assertNull(FileOps.getNodeValue(xmlPath.toString(), "uses-permission", "a:name"));
+        FileOps.addNode(xmlPath, "uses-permission", Map.of("a:name", "android.permission.INTERNET"));
+        assertEquals("android.permission.INTERNET", FileOps.getNodeValue(xmlPath.toString(), "uses-permission", "a:name"));
+        System.out.println(Files.readString(xmlPath));
+        Files.deleteIfExists(xmlPath);
+    }
+    
+    @Test
+    void testPath() throws IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=5533165)